lib/node_modules/@stdlib/plot/table/unicode/lib/props/headers/set.js

```diff
@@ -0,0 +1,63 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2024 The Stdlib Authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+// MODULES //
+
+var format = require( '@stdlib/string/format' );
+var isCollection = require( '@stdlib/assert/is-collection' );
+var isEqual = require( '@stdlib/assert/deep-equal' );
+var parseHeaders = require( './../../parser/parse_headers.js' );
+
+
+// MAIN //
+
+/**
+* Sets the table headers.
+*
+* @private
+* @param {Collection} headers - table headers
+* @throws {TypeError} must be a collection
+* @throws {RangeError} must provide correct number of columns
+*/
```

Comment:
is `Collection` the right type for headers? In most cases, it would be an array, but to include typed arrays I went with `Collection`.

Reply:
Collection is the right choice. Any array-like object should be acceptable.

Reply:
The only thing bothering me is how `{ 'length': 10 }` is also a collection and would pass the initial validations. But accessing any element might break things..

Reply:
Not sure I follow. Why would things break?

Reply:
> { 'length': 10 }

This is equivalent to new Array( 10 ) without proto methods.

Reply:
Sorry, I got confused..